src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -380,24 +246,28 @@ public void updateOdometry() {
       }
 
       // An array list of poses returned by different cameras
-      ArrayList<EstimatedRobotPose> estimatedPoses = m_vision.getEstimatedPoses(m_poseEstimator.getEstimatedPosition());
+      ArrayList<EstimatedRobotPose> estimatedPoses = 
+        m_vision.getEstimatedPoses(m_poseEstimator.getEstimatedPosition());
       // The current position as a translation
-      Translation2d currentEstimatedPoseTranslation = m_poseEstimator.getEstimatedPosition().getTranslation();
+      Translation2d currentEstimatedPoseTranslation = 
+        m_poseEstimator.getEstimatedPosition().getTranslation();
       for (int i = 0; i < estimatedPoses.size(); i++) {
         EstimatedRobotPose estimatedPose = estimatedPoses.get(i);
         // The position of the closest april tag as a translation
         Translation2d closestTagPoseTranslation = new Translation2d();
         for (int j = 0; j < estimatedPose.targetsUsed.size(); j++) {
           // The position of the current april tag
-          Pose3d currentTagPose = m_vision.getTagPose(estimatedPose.targetsUsed.get(j).getFiducialId());
+          Pose3d currentTagPose = m_vision.getTagPose(
+            estimatedPose.targetsUsed.get(j).getFiducialId());
           // If it can't find the april tag's pose, don't run the rest of the for loop for this tag
-          if (currentTagPose == null) {
-            continue;
-          }
+          if (currentTagPose == null) continue;
+
           Translation2d currentTagPoseTranslation = currentTagPose.toPose2d().getTranslation();
-          
-          // If the current april tag position is closer than the closest one, this makes makes it the closest
-          if (j == 0 || currentEstimatedPoseTranslation.getDistance(currentTagPoseTranslation) < currentEstimatedPoseTranslation.getDistance(closestTagPoseTranslation)) {
+
+          // If the current april tag position is closer than the closest one, this makes it the closest
+          if (j == 0 || 
```

Comment:
closestTagPoseTranslation starts off as (0, 0) then when the first index happens is set to that (by checking if j==0) but before that there is `if (currentTagPose == null) continue;` so theoretically if the first one is null and the pose is closer to (0,0) there could be issues. I would suggest making closestTagPoseTranslation = null; and then instead of j == 0, do closestTagPoseTranslation == null.

Reply:
this ijust supose ot be formating maybe add this to the grid alignment PR